operator/builtin/transformer/retain/retain_test.go

```diff
@@ -0,0 +1,322 @@
+package retain
+
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/testutil"
+)
+
+type testCase struct {
+	name      string
+	expectErr bool
+	op        *RetainOperatorConfig
+	input     func() *entry.Entry
+	output    func() *entry.Entry
+}
+
+func TestBuildAndProcess(t *testing.T) {
+	newTestEntry := func() *entry.Entry {
+		e := entry.New()
+		e.Timestamp = time.Unix(1586632809, 0)
+		e.Body = map[string]interface{}{
+			"key": "val",
+			"nested": map[string]interface{}{
+				"nestedkey": "nestedval",
+			},
+		}
+		return e
+	}
+
+	cases := []testCase{
```

Comment:
Looks good. Let's have a test that retains a non-existant field, to validate that it behaves as expected.

Reply:
So, if you give a body field that doesn't actually exist it still treats it as a body field and just retains nothing in the body field and deletes everything. is that how we want it to work? @djaglowski. 

the config and the entry that results from it, if you were curious
```
	{
			"retain_a_non_existant_key",
			false,
			func() *RetainOperatorConfig {
				cfg := defaultCfg()
				cfg.Fields = append(cfg.Fields, entry.NewResourceField("aNonExsistentKey"))
				return cfg
			}(),
			newTestEntry,
			func() *entry.Entry {
				e := newTestEntry()
				return e
			},
		},
```

Reply:
That behavior seems fine to me. Having the test will still be good so that we know if the behavior unexpectedly changes in the future.